### PR TITLE
Lb2 miscellaneous fixes

### PR DESCRIPTION
**Pull request by wprzytula**
To be merged first.

### CODE REVIEW

**Review by havaker — APPROVED** [review]

File: scylla/src/transport/load_balancing/default.rs [hunk 1]
```diff
@@ -36,6 +37,22 @@ impl fmt::Debug for DefaultPolicy {
...
+        if let Some(ref token_with_strategy) = routing_info.token_with_strategy {
+            if self.preferred_datacenter.is_some()
+                && !self.permit_dc_failover
+                && matches!(
+                    token_with_strategy.strategy,
+                    Strategy::SimpleStrategy { .. }
+                )
+            {
+                warn!("\
+Combining SimpleStrategy with preferred_datacenter set to Some and disabled datacenter failover may lead to empty query plans for some tokens.\
```
Comment (havaker) [inline scylla/src/transport/load_balancing/default.rs:49]: I'm worried that it might spam user logs, but idk how to fix this.

Comment (havaker) [reply scylla/src/transport/load_balancing/default.rs:49]: I think it would be good to also add this note to the documentation book.

Comment (havaker) [reply scylla/src/transport/load_balancing/default.rs:49]: But it can be done as a follow up.

Comment (wprzytula) [reply scylla/src/transport/load_balancing/default.rs:49]: As @piodul suggested, this could be another case where log rate limitting would be beneficial.

**Pull request merged**